src/main/java/com/hms_networks/americas/sc/extensions/util/RawNumberValueUtils.java

```diff
@@ -0,0 +1,313 @@
+ *   <li>{@link Float}
+ *   <li>{@link Double}
+ *   <li>{@link Short}
+ *   <li>{@link Byte}
+ * </ul>
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.15.7
+ */
+public class RawNumberValueUtils {
+
+  /**
+   * Get and return the minimum value of the two specified values.
+   *
+   * @param value1 first value
+   * @param value2 second value
+   * @return minimum value
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueMin(Object value1, Object value2) {
+    // Compare values by type
+    Object minValue;
+    if (value1 instanceof Integer && value2 instanceof Integer) {
+      // Compare values
+      Integer int1 = (Integer) value1;
+      Integer int2 = (Integer) value2;
+      if (int1.intValue() < int2.intValue()) {
+        minValue = int1;
+      } else {
+        minValue = int2;
+      }
+    } else if (value1 instanceof Long && value2 instanceof Long) {
+      // Compare values
+      Long long1 = (Long) value1;
+      Long long2 = (Long) value2;
+      if (long1.longValue() < long2.longValue()) {
+        minValue = long1;
+      } else {
+        minValue = long2;
+      }
+    } else if (value1 instanceof Float && value2 instanceof Float) {
+      // Compare values
+      Float float1 = (Float) value1;
+      Float float2 = (Float) value2;
+      if (float1.floatValue() < float2.floatValue()) {
+        minValue = float1;
+      } else {
+        minValue = float2;
+      }
+    } else if (value1 instanceof Double && value2 instanceof Double) {
+      // Compare values
+      Double double1 = (Double) value1;
+      Double double2 = (Double) value2;
+      if (double1.doubleValue() < double2.doubleValue()) {
+        minValue = double1;
+      } else {
+        minValue = double2;
+      }
+    } else if (value1 instanceof Short && value2 instanceof Short) {
+      // Compare values
+      Short short1 = (Short) value1;
+      Short short2 = (Short) value2;
+      if (short1.shortValue() < short2.shortValue()) {
+        minValue = short1;
+      } else {
+        minValue = short2;
+      }
+    } else if (value1 instanceof Byte && value2 instanceof Byte) {
+      // Compare values
+      Byte byte1 = (Byte) value1;
+      Byte byte2 = (Byte) value2;
+      if (byte1.byteValue() < byte2.byteValue()) {
+        minValue = byte1;
+      } else {
+        minValue = byte2;
+      }
+    } else {
+      // Throw exception
+      throw new IllegalArgumentException(
+          "Cannot compare values of type "
+              + value1.getClass().getName()
+              + " and "
+              + value2.getClass().getName());
+    }
+
+    // Return min value
+    return minValue;
+  }
+
+  /**
+   * Get and return the maximum value of the two specified values.
+   *
+   * @param value1 first value
+   * @param value2 second value
+   * @return maximum value
+   * @throws IllegalArgumentException if values are not of the same type
```

Comment:
or are not types that support comparison

Reply:
https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/pull/81#discussion_r1385311719